python/cog/coder/json_coder.py

```diff
@@ -7,7 +7,9 @@
 class JsonCoder(api.Coder):
     @staticmethod
     def factory(cls: Type) -> Optional[api.Coder]:
-        if typing.get_origin(cls) is dict is dict and typing.get_args(cls)[0] is str:
+        if (typing.get_origin(cls) is dict and typing.get_args(cls)[0] is str) or (
+            issubclass(cls, dict)
```

Comment:
Are all sub classes of `dict` `TypedDict`? Is there a more specific way of checking it? Also we need to `decode` a JSON back as `cls` not just a plain `dict[str, Any]`, so we probably need something like `DataclassCoder` that memorizes `cls`.

Reply:
Talked IRL. So `TypedDict` is just a type hint like traffic lights in Florida 🤷, so yeah we can reuse the JSON coder.